examples/server/server.cpp

```diff
@@ -665,30 +665,18 @@ int main(int argc, const char ** argv) {
     svr->wait_until_ready();
     fprintf(stdout, "%s: HTTP server is listening, hostname: %s, port: %d, http threads: %d\n", __func__, args.get_string_param("--host").c_str(), *args.get_int_param("--port"), *args.get_int_param("--n-http-threads"));
 
-    // load the model
-    fprintf(stdout, "%s: loading model and initializing main loop\n", __func__);
 
-    // It might make sense in the long run to have the primary thread run clean up on the response map and keep the model workers parallel.
-    // pool = initialize_workers(args, tqueue, rmap);
     pool = new worker_pool;
-    for (int i = *args.get_int_param("--n-parallelism"); i > 0; i--) {
-        if (i == 1) {
-            fprintf(stdout, "%s: server is listening on http://%s:%d\n", __func__, args.get_string_param("--host").c_str(), *args.get_int_param("--port"));
-            worker * w = new worker(tqueue, rmap, args.get_string_param("--text-encoder-path"), *args.get_int_param("--timeout"));
-            state.store(READY);
-            pool->push_back(w);
-            init_worker(args.get_string_param("--model-path"), *args.get_int_param("--n-threads"), !args.get_bool_param("--use-metal"), default_generation_config, w);
-        } else {
-            worker * w = new worker(tqueue, rmap, args.get_string_param("--text-encoder-path"), *args.get_int_param("--timeout"));
-            w->thread = new std::thread(init_worker, args.get_string_param("--model-path"), *args.get_int_param("--n-threads"), !args.get_bool_param("--use-metal"), default_generation_config, w);
-            pool->push_back(w);
-        }
-    }
+    shutdown_handler = [&](int) {
+        // this should unblock the primary thread;
+        terminate(pool);
```

Comment:
Seeing as `#terminate` joins threads it is probably not safe to perform `#terminate` via the system handler. Will fix.